### PR TITLE
Add migration trial plan selector

### LINKED ISSUE #79526: Add Migration Trial plan selector
Add a new selector in state > sites > plans > selectors that mimics `isECommerceTrialExpired` and a specific `#isSiteOnMigrationTrial()` unit test.

See https://github.com/Automattic/wp-calypso/pull/79673

### PR DESCRIPTION
**Pull request by zaerl**
Fixes https://github.com/Automattic/wp-calypso/issues/79526

## Proposed Changes

* Moved `client/state/sites/plans/selectors/*ecommerce-trial*` to a separate `trials` folder
* Added `is/get-migration-trial days-left/expiration/expired/` files to mimic Woo Express to the new folder
* Split the `client/state/sites/plans/selectors/index.js` in two and added migration unit tests

## Testing Instructions

This PR does not add new UI to the final user. Running:

```
yarn run test-client client/state/sites/plans/test/actions.js
yarn run test-client client/state/sites/plans/test/reducer.js
yarn run test-client client/state/sites/plans/test/selector.js
yarn run test-client client/state/sites/plans/test/trials.js
```

is enough. The tests force `plans/migration-trial` feature on and off.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/state/sites/plans/test/trials.js [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+		const siteId = 1337;
+		test( 'Should return true when the e-commerce trial is in the purchases list', () => {
+			const plan = {
+				ID: 1,
+				productSlug: PLAN_ECOMMERCE_TRIAL_MONTHLY,
+				blogId: siteId,
+				currentPlan: true,
+			};
+
+			const state = getState( plan, siteId );
```
Comment (ouikhuan) [inline client/state/sites/plans/test/trials.js]: Small nitpick here, this plan setup seems to be appearing multiple times in the file, maybe we can turn this into a helper function  by passing `siteId` and `productSlug` into it?

**Comment by ouikhuan** [comment]
I'm seeing several ESLint errors showing up
Screen Shot 2023-07-25 at 11 42 24 AM

**Comment by zaerl** [comment]
> I'm seeing several ESLint errors showing up Screen Shot 2023-07-25 at 11 42 24 AM

I've changed it and made that minor rework of ecommerce/trial tests. Probably the auto-lint made during `git push` isn't 100% working because it doesn't catch these errors. But `yarn run lint:js` instead yes.

**Review by ouikhuan — APPROVED** [review]
LGTM 🚢

**Pull request merged**